shipyard_blueprints/s3/shipyard_s3/s3.py

```diff
@@ -1,39 +1,173 @@
-from shipyard_templates import CloudStorage
 import boto3
+import shipyard_bp_utils as shipyard
+import sys
+import os
+from shipyard_templates import CloudStorage, ShipyardLogger
+from typing import Optional, Dict, Any, List
+from shipyard_s3.utils.exceptions import (
+    DownloadError,
+    InvalidCredentials,
+    MoveError,
+    RemoveError,
+    UploadError,
+)
+from shipyard_s3.utils import utils
+
+
+logger = ShipyardLogger.get_logger()
 
 
 class S3Client(CloudStorage):
     def __init__(
-        self, aws_access_key: str, aws_secret_access_key: str, region: str = None
+        self,
+        aws_access_key: str,
+        aws_secret_access_key: str,
+        region: Optional[str] = None,
     ) -> None:
         self.aws_access_key = aws_access_key
         self.aws_secret_access_key = aws_secret_access_key
         self.region = region
-        super().__init__(
-            aws_access_key=aws_access_key,
-            aws_secret_access_key=aws_secret_access_key,
-            region=region,
-        )
 
     def connect(self):
-        client = boto3.client(
-            "s3",
-            region_name=self.region,
-            aws_access_key_id=self.aws_access_key,
-            aws_secret_access_key=self.aws_secret_access_key,
-            config=None,
-        )
-        self.logger.info("Successfully connected to S3")
-        return client
+        try:
+            logger.debug("Establishing connection with S3")
+            client = boto3.client(
+                "s3",
+                region_name=self.region,
+                aws_access_key_id=self.aws_access_key,
+                aws_secret_access_key=self.aws_secret_access_key,
+                config=None,
+            )
+        except Exception as e:
```

Comment:
may want to include logger.authtest for any useful information we would want the customer to read on a failed connection. This will help customers troubleshoot their credentials issues. This may require handling more specific exceptions raised by s3 depending how readable their errors are